### DISCUSSION TITLE
Requirement of rules_cc v0.0.13 in double-conversion

### DISCUSSION THREAD

**Discussion by tareqsiraj** [bazelbuild/bazel-central-registry#3146] [Q&A] [answered]
Not sure if this is the right place to ask but is there a reason `rules_cc` `v0.0.13` is required to build `double-conversion`? It seems to build fine with an older version (`0.0.9`) in my case with a simple `http_archive` rule.

```bzl
http_archive(
    name = "double-conversion",
    sha256 = "04ec44461850abbf33824da84978043b22554896b552c5fd11a9c5ae4b4d296e",
    strip_prefix = "double-conversion-3.3.0",
    urls = ["https://github.com/google/double-conversion/archive/refs/tags/v3.3.0.tar.gz"],
)
```

https://github.com/bazelbuild/bazel-central-registry/blob/119eb31581bcbf7cde760efe444384353d177a43/modules/double-conversion/3.3.0/patches/module_dot_bazel.patch#L12

@BYVoid

**Comment by zaucy** [comment]
The authors probably simply used the latest at the time. You shouldn't have an issue using 0.0.13 as there shouldn't have been a breaking change afaik

**Comment by BYVoid** [comment, marked as answer]
Yes I simply chose the latest version. Is there a reason why 0.0.13 does not work for you?

**Reply by tareqsiraj** [reply]
I think our toolchain configuration has issues with `rules_cc/0.0.13`, potentially due to shenanigans on our side. For now I'm using the `http_archive` rule to avoid the issue.

In general, is the `bazel_dep` for `rules_cc` required? Or should it be left to the project settings? Some of the other modules in the repository doesn't seem to mention it (i.e. https://github.com/bazelbuild/bazel-central-registry/blob/main/modules/googletest/1.15.2/MODULE.bazel). Thoughts?

**Reply by BYVoid** [reply]
In general, no.

But https://github.com/google/double-conversion/blob/master/BUILD has `load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")`. So `rules_cc` must be added as a dep.

You can add a bazel central registry pr. Patch the `BUILD` to remove that line, and remove the dep from `MODULE.bazel`.

Thanks.

**Reply by BYVoid** [reply]
Done in upstream https://github.com/google/double-conversion/pull/235
